src/main/java/io/cryostat/recordings/RecordingMetadataManager.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright The Cryostat Authors
+ *
+ * The Universal Permissive License (UPL), Version 1.0
+ *
+ * Subject to the condition set forth below, permission is hereby granted to any
+ * person obtaining a copy of this software, associated documentation and/or data
+ * (collectively the "Software"), free of charge and under any and all copyright
+ * rights in the Software, and any and all patent rights owned or freely
+ * licensable by each licensor hereunder covering either (i) the unmodified
+ * Software as contributed to or provided by such licensor, or (ii) the Larger
+ * Works (as defined below), to deal in both
+ *
+ * (a) the Software, and
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software (each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.cryostat.recordings;
+
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Future;
+
+import io.cryostat.core.log.Logger;
+import io.cryostat.core.sys.FileSystem;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import org.apache.commons.lang3.tuple.Pair;
+
+public class RecordingMetadataManager {
+
+    private final Path recordingMetadataDir;
+    private final FileSystem fs;
+    private final Gson gson;
+    private final Logger logger;
+
+    private final Map<Pair<String, String>, String> recordingLabelsMap;
```

Comment:
This `recordingLabelsMap` stores the labels for a recording as a stringified JSON object. All of my unit tests are passing, however if I try to manually start a recording with labels, I can see this `recordingLabelsMap` was updated with the labels from the `201` response. A subsequent GET recordings/ shows the labels for that recording are empty (`labels: ""`). I also tried storing the labels as a `Map<String, String>` with the same curl output.

This is my latest attempt. Do you have any ideas as to why the map value becomes empty?
```
jalaw@localhost ~/cryostat-web (recording-labels)> curl -v --insecure -F recordingName=test -F events=template=ALL -F duration=10 -F labels="{\"KEY\":\"VALUE\",\"key.2\":\"some.value\",\"key3\":\"1234\"}" http://localhost:8181/api/v1/targets/localhost:0/recordings

Trying ::1:8181...
Connected to localhost (::1) port 8181 (#0)
POST /api/v1/targets/localhost:0/recordings HTTP/1.1
Host: localhost:8181
User-Agent: curl/7.76.1
Accept: /
Content-Length: 503
Content-Type: multipart/form-data; boundary=------------------------07df1e96a15e09f6

We are completely uploaded and fine
Mark bundle as not supporting multiuse
< HTTP/1.1 201 Created
< location: /test
< content-type: application/json
< content-length: 472
<
Connection #0 to host localhost left intact
{"downloadUrl":"[http://localhost:8181/api/v1/targets/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Flocalhost:0%2Fjmxrmi/recordings/test",
"reportUrl":"http://localhost:8181/api/v1/targets/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Flocalhost:0%2Fjmxrmi/reports/test",
"labels":"{\"KEY\":\"VALUE\",\"key.2\":\"some.value\",\"key3\":\"1234\"}",
"id":1,"name":"test","state":"RUNNING","startTime":1645125006896,"duration":10000,"continuous":false,"toDisk":true,"maxSize":0,"maxAge":0}⏎](http://localhost:8181/api/v1/targets/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Flocalhost:0%2Fjmxrmi/recordings/test%22,%22reportUrl%22:%22http://localhost:8181/api/v1/targets/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Flocalhost:0%2Fjmxrmi/reports/test%22,%22labels%22:%22%7B/%22KEY/%22:/%22VALUE/%22,/%22key.2/%22:/%22some.value/%22,/%22key3/%22:/%221234/%22%7D%22,%22id%22:1,%22name%22:%22test%22,%22state%22:%22RUNNING%22,%22startTime%22:1645125006896,%22duration%22:10000,%22continuous%22:false,%22toDisk%22:true,%22maxSize%22:0,%22maxAge%22:0%7D%E2%8F%8E) jalaw@localhost ~/cryostat-web (recording-labels)> curl -v --insecure http://localhost:8181/api/v1/targets/localhost:0/recordings
Trying ::1:8181...
Connected to localhost (::1) port 8181 (#0)
GET /api/v1/targets/localhost:0/recordings HTTP/1.1
Host: localhost:8181
User-Agent: curl/7.76.1
Accept: /

Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< content-type: application/json
< content-length: 412
<
Connection #0 to host localhost left intact
[{"downloadUrl":"[http://localhost:8181/api/v1/targets/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Flocalhost:0%2Fjmxrmi/recordings/test",
"reportUrl":"http://localhost:8181/api/v1/targets/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Flocalhost:0%2Fjmxrmi/reports/test",
"labels":"","id":1,"name":"test","state":"STOPPED","startTime":1645125006896,"duration":10000,"continuous":false,"toDisk":true,"maxSize":0,"maxAge":0}]](http://localhost:8181/api/v1/targets/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Flocalhost:0%2Fjmxrmi/recordings/test%22,%22reportUrl%22:%22http://localhost:8181/api/v1/targets/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Flocalhost:0%2Fjmxrmi/reports/test%22,%22labels%22:%22%22,%22id%22:1,%22name%22:%22test%22,%22state%22:%22STOPPED%22,%22startTime%22:1645125006896,%22duration%22:10000,%22continuous%22:false,%22toDisk%22:true,%22maxSize%22:0,%22maxAge%22:0%7D]%E2%8F%8E)
```